app/src/main/java/org/wikipedia/talk/TalkTopicsActivity.kt

```diff
@@ -232,9 +247,23 @@ class TalkTopicsActivity : BaseActivity() {
         binding.talkErrorView.visibility = View.GONE
         binding.talkEmptyContainer.visibility = View.GONE
 
-        disposables.add(ServiceFactory.get(pageTitle.wikiSite).getLastModified(pageTitle.prefixedText)
+        disposables.add(if (resolveTitleRequired) { ServiceFactory.get(pageTitle.wikiSite).getPageNamespaceWithSiteInfo(pageTitle.prefixedText) } else { Observable.just(MwQueryResponse()) }
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
+                .flatMap { response ->
+                    resolveTitleRequired = false
+                    response.query?.namespaces?.let { namespaces ->
+                        response.query?.firstPage()?.let { page ->
+                            val newNs = namespaces.values.find { it.id == page.namespace().code() + 1 }
+                            if (page.namespace().code() % 2 == 0 && newNs != null) {
+                                pageTitle.namespace = newNs.name
+                            }
```

Comment:
Would it be possible to add some comments here to explain the logic?

Reply:
Sure! I agree this is a little too much voodoo without adequate comments.